.github/workflows/ci.yml

```diff
@@ -252,7 +249,13 @@ jobs:
           name: test-results-${{ github.sha }}-all
           path: test-results-${{ github.sha }}-*
           retention-days: 90  # default: 90
+
       - name: Test Summary
         uses: test-summary/action@v1
         with:
           paths: ./test-results-${{ github.sha }}-**/test-report*.xml
+
+      - name: Decide whether the needed jobs succeeded or failed
+        uses: re-actors/alls-green@release/v1
+        with:
+          jobs: ${{ toJSON(needs) }}
```

Comment:
This will always be true since the `needs` condition will prevent the `analyze` job from running if anything failed.

Reply:
Even with the `if: always()` condition above?

Reply:
I've following the example on https://github.com/re-actors/alls-green#usage, e.g. https://github.com/pyca/cryptography/blob/main/.github/workflows/ci.yml#L518-L531

Reply:
https://github.com/orgs/community/discussions/26733#discussioncomment-3253155 has some background

Reply:
oh I see, the analysis job is always run, https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-not-requiring-successful-dependent-jobs

so we want the analysis job to run but "fail" if any of the dependencies also failed?

Reply:
Yeah, basically